### DISCUSSION TITLE
How to: add an alarm to an event?

### DISCUSSION THREAD

**Discussion by firasm** [ics-py/ics-py#277] [Q&A] [answered]
Hey there, 

Sorry to bug with you a silly question, but if I'm able to figure this out, I will happily submit a PR to add this to the documentation. 

I am trying to add an alarm to an event, but am having a hard time figuring out how to actually set it. 

Let's take your [default code to create a simple ics file with a single event](https://icspy.readthedocs.io/en/stable/index.html):

```
from ics import Calendar, Event
c = Calendar()
e = Event()
e.name = "My cool event"
e.begin = '2021-02-01 00:00:00'
c.events.add(e)
c.events
# [<Event 'My cool event' begin:2014-01-01 00:00:00 end:2014-01-01 00:00:01>]
with open('my.ics', 'w') as my_file:
    my_file.writelines(c)
```

If I were to add an alarm to this event, the docs say there's an alarms parameter for each event:

```
alarms (Optional[Iterable[BaseAlarm]]) –
```

So I know I need to specify it like `e.alarms = ??` but it's the end part that I'm struggling with. Through some exporting of blank ics files, I was able to figure out that "2 hours before the event" can be specified as `TRIGGER:-PT2H` 

Thanks in advance!

**Comment by N-Coder** [comment, marked as answer]
Have you tried creating a simple instance of an appropriate alarm? The interface is documented [here](https://icspy.readthedocs.io/en/stable/api.html#alarms), also check out the description of the constructor parameters. Then put your event into a list (which is an "iterable") and use that for your events variable.
Something like the following might work:
```
e.alarms = [DisplayAlarm(trigger=timedelta(hours=-2))]
```
That should also be what you get when you parse an event with the trigger time you mentioned in your post.

An alternative way of answering "how do I replicate XY with ics.py" is creating an appropriate event with your favourite calendaring GUI, exporting it to ics, and checking which objects ics.py gives you when you parse that file. You can then simply create the same objects programmatically and change them as needed.

**Comment by firasm** [comment]
Thanks, I solved the issue!
